### PR TITLE
feat(wakunode2): simplify wakunode2 config and decouple peer persistence

### LINKED ISSUE #1103: store: simplify configuration options
## Background

Currently `store` configuration is quite confusing:
`--store` needs to be enabled, even if a node is only a store client
`--persist-messages` needs to be enabled to store historical messages as a store service node, even if messages are not persisted in DB
`--db-path` is required to actually persist messages on disk

See [this guide](https://github.com/status-im/nwaku/blob/ad7f9ed5bfe5c47cd581149a20def03d019b82a6/docs/operators/how-to/configure-store.md#configuring-a-store-client) for more.

## Suggested solution

It makes much more sense to me this way: `store` should in fact only be mounted if you are storing messages yourself and this implies:
- only the `--store` config option is necessary (the confusing `--persist-messages` can be removed <-- this can probably be done in any case)
- store query API methods should always be available, as it's unrelated to whether `store` is mounted or not.

This has been discussed before in the context of [advertising store protocol](https://github.com/status-im/go-waku/issues/292#issuecomment-1227474687) and [handling asymmetrical protocols in libp2p](https://github.com/status-im/nim-libp2p/issues/568)

### PR DESCRIPTION
**Pull request by LNSD**
This is the second part of #1293, and resolves #1103:

- [x] Decouple peer persistence database from message store database
- [x] Decouple `--storenode` from `resumeStoreNode`
- [x] Simplified `wakunode2` storage setup procedures
- [x] Made `SqliteDatabase` compatible with `dbUrl` path
- [x] Removed the deprecated configuration options from `wakunode2/config.nim`
- [x] Updated the tests accordingly

**NOTE:** Some extra work is necessary to improve the message store options (e.g., implement the unimplemented methods in the in-memory store, decouple the message store from the waku store protocol, etc.). This will be done in future PRs.

### CODE REVIEW

File: apps/wakunode2/wakunode2.nim
Comment (rymnc) [inline apps/wakunode2/wakunode2.nim:79]:
```suggestion
    return err("failed to init database connection: " & connRes.error)
```

File: apps/wakunode2/wakunode2.nim
Comment (rymnc) [inline apps/wakunode2/wakunode2.nim:142]: Should we use an optional enum for the retentionPolicy argument since we have only 2, or *no* retention policy at all?

File: apps/wakunode2/wakunode2.nim
Comment (rymnc) [inline apps/wakunode2/wakunode2.nim:189]:
```suggestion
    if migrate:
      ?performDbMigration(db.get(), migrationPath=MessageStoreMigrationPath)
```

File: apps/wakunode2/wakunode2.nim
Comment (rymnc) [inline apps/wakunode2/wakunode2.nim:523]: a try-except here maybe?

**Review by rymnc — APPROVED** [review]
LGTM 😁

**Review by jm-clius — APPROVED** [review]
LGTM! LMK if you want to divide and conquer the subsequent tasks (infra updates, doc updates, etc.)

File: apps/wakunode2/wakunode2.nim
Comment (LNSD) [reply apps/wakunode2/wakunode2.nim:142]: Can you elaborate on that? Where are you suggesting using the enum?

- Return type?
- Input argument?
- Config?

File: apps/wakunode2/wakunode2.nim
Comment (LNSD) [reply apps/wakunode2/wakunode2.nim:523]: Well... the compiler is not complaining, so I don't think we should add a try/catch there.

But I agree that the error management in the metrics, JSON-RPC, and rest servers setup procedure has to be improved.

**Comment by LNSD** [comment]
> LGTM! LMK if you want to divide and conquer the subsequent tasks (infra updates, doc updates, etc.)

@jakubgs In this PR, we are dropping the deprecated options. So in the next release, v0.13.0, the store configuration will change. Do you have any concerns before merging the PR?

**Comment by jakubgs** [comment]
Yes, we'll need to coordinate the deployment to fleets or shit will break. I'm in Istanbul next week and off most of week after that, so if we want this done it would probably have to be tomorrow. Or on the 9th possibly.

File: apps/wakunode2/wakunode2.nim
Comment (rymnc) [reply apps/wakunode2/wakunode2.nim:142]: in the input, sorry. `retentionPolicy: string`

**Comment by LNSD** [comment]
No need to rush. AFAIK, we are not planning to do any rollout for now. We can wait until you are back. 

Probably it is a better idea because I am planning to add support for configuring via environment variables. So we do not re-work things twice.

**Pull request merged**
